### ISSUE TITLE
Add disallowing declaration after nested block

### ISSUE THREAD

**Issue by indrekpaas** [stylelint/stylelint#3719]
> What is the problem you're trying to solve?

A declaration before or after a nested block will always yield to same output. However, having a declaration after a nested block reduces legibility significantly.

> What solution would you like to see?

A new rule that would disallow declaration after a nested block, e.g.

```scss
/* Violation */
.parent {
  font-family: sans-serif;

  .child {
    width: 100%;
  }

  font-size: 1em;
}

/* No violation */
.parent {
  font-family: sans-serif;
  font-size: 1em;

  .child {
    width: 100%;
  }
}

// The above produces identical results

/* Violation */
.parent {
  font-family: sans-serif;
  font-size: 1em;
}
.parent .child {
  width: 100%;
}

/* No violation */
.parent {
  font-family: sans-serif;
  font-size: 1em;
}
.parent .child {
  width: 100%;
}
```

* Name: `declaration-block-no-declaration-after-nested-block`
* Primary option: `true`
* Secondary options: none
* Message: `Expected no declaration after nested block`
* Section: "Declaration block"

**Comment by hudochenkov** [comment]
For this scenario, the perfect solution is [stylelint-order](https://github.com/hudochenkov/stylelint-order) plugin and it's `order` rule.
